Meraki scenario guide - describe how to merge new data with old data (#48999)
ansible/ansible: docs/docsite/rst/scenario_guides/guide_meraki.rst

.. code-block:: json

    [
        {
            "orgAccess": "full", 
            "name": "John Doe",
            "tags": [],
            "networks": [],
            "email": "john@doe.com",
            "id": "12345677890"
        }
    ]

Handling Returned Data
======================

Since Meraki's response data uses lists instead of properly keyed dictionaries for responses, certain strategies should be used when querying data for particular information. For many situations, use the ``selectattr()`` Jinja2 function.

Merging Existing and New Data
=============================

> Ansible's Meraki modules do not allow for manipulating data. For example, there is likely a need to insert a rule in the middle of a firewall ruleset. Ansible and the Meraki modules lack a way to directly merge to manipulate data. However, a playlist can use a few tasks to split the list where a rule needs to be inserted and then merge them together again with the new rule added. The steps involved are as follows:

Reviewer: ```suggestion
Ansible's Meraki modules do not allow for manipulating data. For example, you may need to insert a rule in the middle of a firewall ruleset. Ansible and the Meraki modules lack a way to directly merge to manipulate data. However, a playlist can use a few tasks to split the list where you need to insert a rule and then merge them together again with the new rule added. The steps involved are as follows:
```

1. Create blank "front" and "back" lists.
    ::

        vars:
          - front_rules: []
          - back_rules: []
2. Get existing firewall rules from Meraki and create a new variable.
    ::

        - name: Get firewall rules
          meraki_mx_l3_firewall:
            auth_key: abc123
            org_name: YourOrg
            net_name: YourNet
            state: query
          delegate_to: localhost
          register: rules
        - set_fact:
            original_ruleset: '{{rules.data}}'
3. Write the new rule.
    ::

        - set_fact:
            new_rule:

>               - 

Reviewer 2: Is this line right?

Author: I thought this posted a comment but apparently it's not there. It's right because I need to merge lists later and it wasn't able to merge a dict and a list together, so I wrapped the dict in a list so the merge works. Is there a better way to do it?

Reviewer 3: @kbreit @gundalow line 151 explains the presence of the `-`

Reviewer 3: d'oh, you just added that line . . . I need more caffeine
